### PR TITLE
fix(npm): Prevents canary releases with double dashed version numbers

### PR DESCRIPTION
**Pull request by 10hendersonm**
# What Changed

Removed leading dash on all `npm` canary releases.

## Why

Since upgrading to `auto@10.x`, our canary releases are creating versions like `10.1.0--canary.1379.2451.0` with a double-dash on the pre-id.

The other instances in the `canary` hook were already stripping off the leading `-`, so I've extended this to the last 2 instances.

Todo:

- [ ] Add tests
- [ ] Add docs

## Change Type

Indicate the type of change your pull request is:

- [ ] `documentation`
- [x] `patch`
- [ ] `minor`
- [ ] `major`

:baby_chick: Download canary assets:

[auto-linux--canary.1997.24321.gz](https://github.com/intuit/auto/releases/download/v0.0.0-canary/auto-linux--canary.1997.24321.gz)  
[auto-macos--canary.1997.24321.gz](https://github.com/intuit/auto/releases/download/v0.0.0-canary/auto-macos--canary.1997.24321.gz)  
[auto-win.exe--canary.1997.24321.gz](https://github.com/intuit/auto/releases/download/v0.0.0-canary/auto-win.exe--canary.1997.24321.gz)

<details>
  <summary>📦 Published PR as canary version: <code>under canary scope @auto-canary@10.29.1-canary.1997.24321.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @auto-canary/bot-list@10.29.1-canary.1997.24321.0
  npm install @auto-canary/auto@10.29.1-canary.1997.24321.0
  npm install @auto-canary/core@10.29.1-canary.1997.24321.0
  npm install @auto-canary/package-json-utils@10.29.1-canary.1997.24321.0
  npm install @auto-canary/all-contributors@10.29.1-canary.1997.24321.0
  npm install @auto-canary/brew@10.29.1-canary.1997.24321.0
  npm install @auto-canary/chrome@10.29.1-canary.1997.24321.0
  npm install @auto-canary/cocoapods@10.29.1-canary.1997.24321.0
  npm install @auto-canary/conventional-commits@10.29.1-canary.1997.24321.0
  npm install @auto-canary/crates@10.29.1-canary.1997.24321.0
  npm install @auto-canary/docker@10.29.1-canary.1997.24321.0
  npm install @auto-canary/exec@10.29.1-canary.1997.24321.0
  npm install @auto-canary/first-time-contributor@10.29.1-canary.1997.24321.0
  npm install @auto-canary/gem@10.29.1-canary.1997.24321.0
  npm install @auto-canary/gh-pages@10.29.1-canary.1997.24321.0
  npm install @auto-canary/git-tag@10.29.1-canary.1997.24321.0
  npm install @auto-canary/gradle@10.29.1-canary.1997.24321.0
  npm install @auto-canary/jira@10.29.1-canary.1997.24321.0
  npm install @auto-canary/magic-zero@10.29.1-canary.1997.24321.0
  npm install @auto-canary/maven@10.29.1-canary.1997.24321.0
  npm install @auto-canary/microsoft-teams@10.29.1-canary.1997.24321.0
  npm install @auto-canary/npm@10.29.1-canary.1997.24321.0
  npm install @auto-canary/omit-commits@10.29.1-canary.1997.24321.0
  npm install @auto-canary/omit-release-notes@10.29.1-canary.1997.24321.0
  npm install @auto-canary/pr-body-labels@10.29.1-canary.1997.24321.0
  npm install @auto-canary/released@10.29.1-canary.1997.24321.0
  npm install @auto-canary/s3@10.29.1-canary.1997.24321.0
  npm install @auto-canary/sbt@10.29.1-canary.1997.24321.0
  npm install @auto-canary/slack@10.29.1-canary.1997.24321.0
  npm install @auto-canary/twitter@10.29.1-canary.1997.24321.0
  npm install @auto-canary/upload-assets@10.29.1-canary.1997.24321.0
  npm install @auto-canary/vscode@10.29.1-canary.1997.24321.0
  # or 
  yarn add @auto-canary/bot-list@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/auto@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/core@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/package-json-utils@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/all-contributors@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/brew@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/chrome@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/cocoapods@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/conventional-commits@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/crates@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/docker@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/exec@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/first-time-contributor@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/gem@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/gh-pages@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/git-tag@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/gradle@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/jira@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/magic-zero@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/maven@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/microsoft-teams@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/npm@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/omit-commits@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/omit-release-notes@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/pr-body-labels@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/released@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/s3@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/sbt@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/slack@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/twitter@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/upload-assets@10.29.1-canary.1997.24321.0
  yarn add @auto-canary/vscode@10.29.1-canary.1997.24321.0
  ```
</details>

### CODE REVIEW

**Comment by 10hendersonm** [comment]
Seeing this in my PR body was pretty funny:

image

I get the feeling my PR doesn't resolve that, as I've only touched `npm`.

**Comment by 10hendersonm** [comment]
I'm also up in the air about whether this is a `patch` or a `major`. If somebody has come to rely on regexes to parse semvers, it's breaking for them, but I also believe it's unintended in the current implementation, meaning it's a fix whether or not it breaks somebody having taken steps to consume it in the broken fashion.

It also seems super minor to rev up to `v11` for just this.

TL;DR IMO it's a `patch`

**Comment by rconnamacher** [comment]
@hipstersmoothie @10hendersonm This was an intentional change to cause canary releases to not override beta releases in standard Semver ranges. A pull request opened by any non-contributor should never match a prerelase version range like `>= 1.0.0-beta.1`. Is that being undone?

**Comment by adierkens** [comment]
:rocket: PR was released in [`v10.29.3`](https://github.com/intuit/auto/releases/tag/v10.29.3) :rocket:
